Content.Server/Projectiles/ProjectileSystem.cs

```diff
@@ -37,7 +37,18 @@ private void OnStartCollide(EntityUid uid, ProjectileComponent component, ref St
 
         var target = args.OtherEntity;
         // it's here so this check is only done once before possible hit
-        var attemptEv = new ProjectileReflectAttemptEvent(uid, component, false);
+	// Moffstation - Begin - Ricochet
+        var attemptEv = new ProjectileReflectAttemptEvent(uid, component, false, target);
+
+        // Possible ricochet -- raise event first on the projectile and then on the target
+        RaiseLocalEvent(uid, ref attemptEv);
+        if (attemptEv.Cancelled)
+        {
+            SetShooter(uid, component, target);
+            return;
+        }
```

Comment:
Instead of reusing this event, can you create a complementary one which contains basically the same stuff, but is meant for this case.
That is to say, the existing event is assumed to only ever be raised on the target; the complementary event would be only ever raised on the projectile.